test/MessageTraitTest.php

```diff
@@ -50,7 +50,6 @@ public function invalidProtocolVersionProvider()
             '1-without-minor'      => [ '1' ],
             '1-with-invalid-minor' => [ '1.2' ],
             '1-with-hotfix'        => [ '1.2.3' ],
-            '2-with-minor'         => [ '2.0' ],
```

Comment:
This used to be explicitly tested/marked invalid: did the RFC around it change?

Reply:
while protocol is called HTTP/2 we do not work with it here and the [ connection preface](https://www.rfc-editor.org/rfc/rfc7540.html#section-3.5) used 


> In HTTP/2, each endpoint is required to send a connection preface as a final confirmation of the protocol in use and to establish the initial settings for the HTTP/2 connection.

> the connection preface starts with the string `PRI * HTTP/2.0\r\n\r\nSM\r\n\r\n`

That implies that 2.0 is a correct value

Reply:
if so, the comment should change as well, not?

Reply:
I did not read any of the spec aside for quick scan. Does it matter outside of server request? `2.0` might need to be normalized to 2, i guess, but it should be accepted if (some) servers are sending it.

Reply:
This is a pretty much stock NGINX install, just added `http2` to the `listen` line and that's how it gets the value from `$_SERVER['SERVER_PROTOCOL']`